scrapyd_k8s/api.py

```diff
@@ -137,6 +143,41 @@ def enable_authentication(app, config_username, config_password):
     app.config["BASIC_AUTH_FORCE"] = True
     return basic_auth
 
+def watch_pods(namespace):
+    num_lines_to_check = scrapyd_config.get('num_lines_to_check')
+    w = watch.Watch()
+    v1 = client.CoreV1Api()
+    for event in w.stream(v1.list_namespaced_pod, namespace=namespace):
+        pod = event['object']
+        # check the labels in the docs
+        if pod.status.phase == 'Running' and pod.metadata.labels.get("org.scrapy.job_id"):
+            thread_name = "%s_%s" % (namespace, pod.metadata.name)
+            if (thread_name in watcher_threads
+                    and watcher_threads[thread_name] is not None
+                    and watcher_threads[thread_name].is_alive()):
+                pass
+            else:
+                watcher_threads[thread_name] = threading.Thread(
+                    target=stream_logs,
+                    kwargs={
+                        'job_name': pod.metadata.name,
+                        'namespace': namespace,
+                        'num_lines_to_check': num_lines_to_check
+                    }
+                )
+                watcher_threads[thread_name].start()
+        elif pod.status.phase == 'Succeeded' and pod.metadata.labels.get("org.scrapy.job_id"):
```

Comment:
What happens if the job errored? Is the phase then also `Succeeded`?
I think we would like to see the logs also when it errored (e.g. the process terminated).

Reply:
If the job errors, then the status of the pod is "Failed". In our schedule function we aim to 1 completion but have  backoff_limit=0, so if a job fails at least 1 time, the pod status will change (the status of pod is changing to failed if at least 1 container inside has a non-zero exit code).

Reply:
I will add code to collect logs from failed pods, need to think a bit if it belongs to the main loop with a happy scenario.